### PR TITLE
Fix issues related to apply ion-test-driver to ion-c and ion-js.

### PR DESCRIPTION
**Pull request by cheqianh**
Fixes issues and test in this branch.

### CODE REVIEW

**Comment by cheqianh** [comment]
Above commit Improved analyze disagree list logic for write_compare phase.

For example: 
`ion-java_1,text,ion-c` means using ion-java_1 to rewrite ion-c in text format. Before we compare this to all the files that re-written by ion-java_2. Now I made it more accurate that `ion-java_1,text,ion-c` only compares to `ion-java_2,text,ion-c`.
By this way we can make sure they behaviors same for a specific action.

For code:
`first_file`/ `second_file` means the file re-written by the first impl/ second imp.
line 860 `agree` variable represents if two disagree list agree with each other, if not alert users.
line 862, `second_file` is the variable represents what the current file will compare to. e.g. `ion-java_1,text,ion-c` only compares to `ion-java_2,text,ion-c`.
line 868 if we find the first file in the second file's disagree list, then they disagree with each other.

**Comment by cheqianh** [comment]
Above commit fixed different message issue.
Since we recognize the error type by message. So if two messages shows the different text we treat it as different errors.
In addition, message text is generated by cli tools.
I noticed that ion-c always include implementations' name in the message which will makes ion-test-driver treat 2 same type issue differently.
To avoid this, and make sure implementation's revision number doesn't affect error type, I removed all revision number. For example ion-c_abcd1234 will be replaced to ion-c.

File: amazon/iontest/ion_test_driver.py [hunk 1]
```diff
@@ -931,6 +921,18 @@ def validate_results(report, result_field, read_error, read_compare, write_error
...
+    if '_' not in first_impl or '_' not in second_impl or 'message' not in obj.keys():
+        return
+    obj['message'] = obj['message'].replace(first_impl, first_impl.split('_')[0])
+    obj['message'] = obj['message'].replace(second_impl, second_impl.split('_')[0])
+
+
+def replace_impl_name_for_obj(errors, first_impl, second_impl):
+    for error in errors:
+        replace_impl_name_for_message(error, first_impl, second_impl)
+
```
Comment (tgregg) [inline amazon/iontest/ion_test_driver.py:934]: Please add comments in the code to explain why these are needed. The description in the PR makes sense, but it is a good idea to have a similar explanation in the code to make it easier to follow.

**Review by tgregg — APPROVED** [review]

File: amazon/iontest/ion_test_driver.py (hunk 1 above)
Comment (cheqianh) [reply amazon/iontest/ion_test_driver.py]: Added.

**Pull request merged**